src/zope/interface/exceptions.py

```diff
@@ -97,19 +117,49 @@ class BrokenMethodImplementation(Invalid, _TargetMixin):
     """
 
     def __init__(self, method, mess, target=_NotGiven):
-        Invalid.__init__(self)
+        Invalid.__init__(self, method, mess, target)
         self.method = method
         self.mess = mess
         self.target = target
 
-    def __str__(self):
-        return "%s violates its contract in %s: %s." % (
-            self._prefix,
+    @property
+    def _specifics(self):
+        return 'violates the contract of %s because %s' % (
             repr(self.method) if isinstance(self.method, str) else self.method,
-            self.mess
+            self.mess,
         )
 
 
+class MultipleInvalid(_TargetMixin, Invalid):
+    """
+    The *target* has failed to implement the *iface* in
+    multiple ways.
+
+    The failures are described by *exceptions*, a collection of
+    other `Invalid` instances.
+
+    .. versionadded:: 5.0
+    """
+
+    def __init__(self, iface, target, exceptions):
+        exceptions = list(exceptions)
+        Invalid.__init__(self, iface, target, exceptions)
+        self.target = target
+        self.interface = iface
+        self.exceptions = exceptions
+
+    @property
+    def _specifics(self):
+        # It would be nice to use tabs here, but that
+        # is hard to represent in doctests.
+        return '\n    ' + '\n    '.join(
+            x._specifics.strip() if isinstance(x, _TargetMixin) else(str(x))
```

Comment:
No need to wrap `(str(x))` in parens.

Reply:
I blame an overaggresive [smartparens](https://github.com/Fuco1/smartparens).